### ISSUE TITLE
Stuck on 'publishing' personal profile with macOS 10.12.6

### ISSUE THREAD

**Issue by silantnode** [ssbc/patchwork#726]
- Installed Patchwork 3.6.8 on macOS 10.12.6
- Launched app
- App displays 'scuttling' in grab bar, 'loading' in the main timeline area and the loading icon in the 'send private message' column
- Clicked on 'edit profile'
- Dialog box appears, and I load in an image and enter text for name and description
- Clicked on 'publish'

Result: 'Publishing...' button remains greyed out and can stay this way forever unless I hit cancel.
If I do hit cancel the app continues its loading, but that makes sense because I am not yet connected to a pub. 

Questions:
- Is there a compatibility problem with macOS 10.12.6 and Patchwork 3.6.8?
- Would the process of entering and saving my personal profile benefit from connecting to a pub? And would that not defeat the purpose of being able to work offline?

**Comment by mmckegg** [comment]
@silantnode 

If you got that message and you **haven't** connected to any pubs yet, then something has gone horribly wrong!

You should be able to access the error if you can start patchwork from the terminal.

```sh
# run this in terminal.app
/Applications/Patchwork.app/Contents/MacOS/Patchwork

# try to create a profile as before
# Error should be logged out
```

**Comment by silantnode** [comment]
Thank you for your response @mmckegg.

After the launch command the terminal printed this following:

```
{ party: true,
  host: '',
  port: 8008,
  timeout: 0,
  pub: true,
  local: true,
  friends: { dunbar: 150, hops: 2 },
  ws: { port: 8989 },
  gossip: { connections: 3 },
  path: '/Users/silant/.ssb',
  timers: { connection: 0, reconnect: 5000, ping: 300000, handshake: 5000 },
  caps: 
   { shs: '1KHLiKZvAvjbY1ziZEHMXawbCEIM6qwjCDm3VYRan/s=',
     sign: null },
  master: [],
  logging: { level: 'notice' },
  blobsPort: 7777,
  server: true,
  _: [] }
Log level: notice
git-ssb is not installed, or not available in path
```
Then the state of the application is the same as described in my initial commend.

Interesting that it is requiring git-ssb. I am ready to do this if necessary.

**Comment by mmckegg** [comment]
Strange, no other messages in the log after the failure?

One other place you can find logs is by right clicking on the patchwork interface and selecting "Inspect Element" then choosing the "Console" tab.

If no errors are being thrown, but the profile never publishes then this is very strange indeed.

---

No, you shouldn't need git-ssb.

**Comment by silantnode** [comment]
No other messages after the failure.

Inspected a handful of elements. Nothing in any of the consoles.  Perhaps there is a specific element I should target @mmckegg?

**Comment by mmckegg** [comment]
@silantnode 

Not really sure what else to suggest, except wiping the data directory and starting again.

Make sure patchwork is closed then open terminal and type:

```shell
rm -rf ~/.ssb
```

Then relaunch patchwork. Maybe it will work this time?

**Comment by silantnode** [comment]
Thank you for the follow up @mmckegg.

I did the following:

- Quit Patchwork
- Opened Terminal and ran rm `-rf ~/.ssb`
- Quit Terminal
- Restarted system
- Launched Patchwork

Result was the same.

Might this [issue](https://github.com/ssbc/patchwork/issues/727) be related?

**Comment by mariosangiorgio** [comment]
I was having the same issue, but I figured out that I was doing something wrong. I was following the documentation at https://scuttlebot.io/ and installed and started `sbot server`.

After that I launched Patchwork and it got stuck. From the logs I noticed that it was trying to use port 8008, which was already in use. I then shut down both the server and Patchwork and restarted only Patchwork and everything works fine now.

I'm not sure this is the same issue you're experiencing, but it hope it helps.

**Comment by silantnode** [comment]
Thank you for the tip @mariosangiorgio. I had not actually installed sbot as I came to Patchwork through the [http://scuttlebutt.nz](http://scuttlebutt.nz) website that does not state this requirement.

@mmckegg the problem may be a conflict with Patchwork and something else installed in my home directory / library as I was able to run Patchwork just fine in a separate user account. It's now just a matter of figuring out what that is.
